test/model/internal_link_test.dart

```diff
@@ -451,4 +452,48 @@ void main() {
       });
     });
   });
+
+  group('narrowLink URL generation', () {
+    final streams = [
+      eg.stream(streamId: 1, name: 'check'),
+    ];
+
+    test('narrowLink should include a trailing slash in the path and before #', () async {
+      final store = await setupStore(realmUrl: realmUrl, streams: streams);
+      final message = eg.streamMessage();
+      final narrow = TopicNarrow.ofMessage(message);
+      final messageLink = narrowLink(store, narrow, nearMessageId: 12345);
+      final urlString = messageLink.toString();
+      final fragmentIndex = urlString.indexOf('#');
+      expect(fragmentIndex, greaterThan(0));
+      if (fragmentIndex > 0) {
+        final lastCharBeforeFragment = urlString[fragmentIndex - 1];
+        expect(lastCharBeforeFragment, equals('/'));
+      }
+    });
+  });
+
+  group('narrowLink Full URL generation', () {
+    final streams = [
+      eg.stream(streamId: 123, name: 'check'),
+    ];
+    test('narrowLink should generate the full URL correctly', () async {
+      final store = await setupStore(realmUrl: realmUrl, streams: streams);
```

Comment:
This `realmUrl` is `https://example.com/`, which comes with a trailing `/`. So we might want to override this for testing.